onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java

```diff
@@ -268,6 +269,23 @@ public View onCreateView(LayoutInflater inflater,
     }
 
     @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        RouteFavoriteDialogFragment dialogFragment = (RouteFavoriteDialogFragment)
+                getFragmentManager().findFragmentByTag(RouteFavoriteDialogFragment.TAG);
+        if (dialogFragment != null) {
```

Comment:
Could you put the code that's a duplicate of lines 721-728 (setting the callback) in a method, and call that method from both locations?  I could see having the duplicate code causing headaches down the road, of someone changing the code in lines 721-728 and then trying to figure out why the behavior isn't changing.